test/pattern.spec.ts

```diff
@@ -0,0 +1,62 @@
+import execa from 'execa'
+import mock from 'mock-fs'
+
+import prettyQuick from 'pretty-quick'
+
+jest.mock('execa')
+
+afterEach(() => {
+  mock.restore()
+  jest.clearAllMocks()
+})
+
+describe('match pattern', () => {
+  it('issue #73 #125 - regex grouping (round pattern)', () => {
+    const onFoundChangedFiles = jest.fn()
+
+    mock({
+      '/.git': {},
+      '/src/apps/hello/foo.js': "export const foo = 'foo'",
+      '/src/libs/hello/bar.js': "export const bar = 'bar'",
+      '/src/tools/hello/baz.js': "export const baz = 'baz'",
+      '/src/should-not-be-included/hello/zoo.js': "export const zoo = 'zoo'",
+    })
+
+    // @ts-expect-error -- Need to find a better way to moch this
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    execa.sync.mockImplementation((_command: string, args: string[]) => {
```

Comment:
Can we destruct `args` into more meaningful names here?

Reply:
Hmmm, unlikely. `args` here is indeterministic.

If I want `execa.sync` to execute `git merge-base --cached`, then the `args` will be `['merge-base', '--cached']`. And if I want `execa.sync` to execute `git --quiet ls-files`, then the `args` will be `['--quiet', 'ls-files']`.

Reply:
Fair enough.